internal/wasm/module_test.go

```diff
@@ -828,3 +829,77 @@ func TestModule_validateDataCountSection(t *testing.T) {
 		}
 	})
 }
+
+func TestModule_declaredFunctionIndexes(t *testing.T) {
```

Comment:
thanks for putting in unit tests for this! can you make a test for each of the main special cases noted in the comment (include test name field please)